= RNA-seq data
:setup_script: /no_backup_isis/rg/projects/courses/rnaseq/setup2015nov

In order to get all the required files for the hands-on you need to run the following command in your terminal:

[source,bash, subs="attributes"]
----
{setup_script}
----

The setup script copies all the required files into the `rnaseq` folder under your `$HOME`. You can see the contents of the directory in a tree-like format as follows:

[source,bash]
----
tree ~/rnaseq
----

And this should be the output:

[source,bash]
----
/users/<group>/<username>/rnaseq
├── bin
│   ├── BAMstats.py
│   ├── dashboard2tsv.py
│   ├── DESeq.analysis.R
│   ├── edgeR.analysis.R
│   ├── gff2gff.awk
│   ├── ggheatmap.R
│   ├── GO_enrichment.R
│   ├── matrix_to_dist.R
│   ├── normalize.matrix.R
│   ├── retrieve_element_rpkms.py
│   ├── rpkm_distribution.R
│   ├── selectMatrixRows.sh
│   ├── terrain.colors.3.txt
│   ├── TrtoGn_RPKM.sh
│   └── VennDiagram.R
├── data
│   ├── mouse_cns_E14_rep1_1.fastq.gz
│   ├── mouse_cns_E14_rep1_2.fastq.gz
│   ├── mouse_cns_E14_rep2_1.fastq.gz
│   ├── mouse_cns_E14_rep2_2.fastq.gz
│   ├── mouse_cns_E18_rep1_1.fastq.gz
│   ├── mouse_cns_E18_rep1_2.fastq.gz
│   ├── mouse_cns_E18_rep2_1.fastq.gz
│   ├── mouse_cns_E18_rep2_2.fastq.gz
│   ├── quantifications.index.tsv
│   └── quantifications.index.txt
└── refs
    ├── mm65.long.ok.gtf
    ├── mouse_genome_mm9.fa.fai
    ├── mouse_genome_mm9_RSEM_index
    │   ├── RSEMref.chrlist
    │   ├── RSEMref.grp
    │   ├── RSEMref.idx.fa
    │   ├── RSEMref.n2g.idx.fa
    │   ├── RSEMref.seq
    │   ├── RSEMref.ti
    │   └── RSEMref.transcripts.fa
    └── mouse_genome_mm9_STAR_index -> /no_backup_isis/rg/projects/courses/rnaseq/refs/mouse_genome_mm9_STAR_index

5 directories, 34 files
----

NOTE: **STAR** index folder has been soft linked because of its size
