= RNA-seq signal files

We user STAR to generate BedGraph files from the alignments. Launch the following command to run it:

[source,cmd,subs="{markup-in-source}"]
----
STAR --runThreadN 2 \
     --runMode inputAlignmentsFromBAM \
     --inputBAMfile alignments/mouse_brain_rep1_Aligned.sortedByCoord.out.bam \
     --outWigType bedGraph \
     --outWigStrand Unstranded \
     --outFileNamePrefix alignments/mouse_brain_rep1_ \
     --outWigReferencesPrefix chr
----
// real	0m0.691s
// user	0m0.673s
// sys	0m0.018s

NOTE: In order to simplify this part of the hands-on we are processing the data as unstranded even if it is [crg]#**stranded**#.

STAR produces two files: one using uniquely mapped reads, the other using multimapped reads. We convert one of the BedGraph files to BigWig:

[source,cmd,subs="{markup-in-source}"]
----
bedGraphToBigWig alignments/mouse_brain_rep1_Signal.Unique.str1.out.bg \
                 {rnaseq_folder}refs/genome.fa.fai \
                 alignments/mouse_brain_rep1_uniq.bw
----
// real	0m0.440s
// user	0m0.425s
// sys	0m0.013s
NOTE: Remember that, in general, you can always have a look at programs options by just typing the command name or `--help` after the command (e.g. `STAR --help`).
