= RNA-seq signal files

We user STAR to generate BedGraph files from the alignments. Launch the following command to run it:

[source,cmd,subs="{markup-in-source}"]
----
STAR --runThreadN 2 \
     --runMode inputAlignmentsFromBAM \
     --inputBAMfile alignments/mouse_cns_E18_rep1_Aligned.sortedByCoord.out.bam \
     --outWigType bedGraph \
     --outWigStrand Unstranded \
     --outFileNamePrefix alignments/mouse_cns_E18_rep1_ \
     --outWigReferencesPrefix chr
----

NOTE: For demonstation purposes in the context of the hands on we are processing the data as unstranded even if it is [crg]#**stranded**#.

STAR produces two files: one using uniquely mapped reads, the other using multimapped reads. We convert one of the BedGraph files to BigWig:

[source,cmd,subs="{markup-in-source}"]
----
bedGraphToBigWig alignments/mouse_cns_E18_rep1_Signal.Unique.str1.out.bg \
                 /rnaseq/refs/genome.fa.fai \
                 alignments/mouse_cns_E18_rep1_uniq.bw
----

