= Mapping
Create a folder for the alignment steps:

[source,cmd]
----
mkdir alignments
----

Create the genome index for STAR:

[source,cmd,subs="{markup-in-source}"]
----
mkdir genomeIndex
----

[source,cmd,subs="{markup-in-source}"]
----
STAR --runThreadN 2 \
     --runMode genomeGenerate \
     --genomeDir genomeIndex \
     --genomeFastaFiles {rnaseq_folder}refs/genome.fa \
     --sjdbGTFfile {rnaseq_folder}refs/anno.gtf \
     --genomeSAindexNbases 11
----
// real	0m20.906s
// user	0m20.811s
// sys	0m0.591s

Run the following command to perform the mapping:

[source,cmd,subs="{markup-in-source}"]
----
STAR --runThreadN 2 \
     --genomeDir genomeIndex \
     --readFilesIn {rnaseq_folder}data/mouse_cns_E18_rep1_1.fastq.gz \
                   {rnaseq_folder}data/mouse_cns_E18_rep1_2.fastq.gz \
     --outSAMunmapped Within \
     --outFilterType BySJout \
     --outSAMattributes NH HI AS NM MD \
     --readFilesCommand pigz -p2 -dc \
     --outSAMtype BAM SortedByCoordinate \
     --quantMode TranscriptomeSAM \
     --outFileNamePrefix alignments/mouse_cns_E18_rep1_
----
// real	0m40.882s
// user	0m40.841s
// sys	0m0.261s

When finished we can look at the bam file:

[source,cmd,subs="{markup-in-source}"]
----
samtools view -h alignments/mouse_cns_E18_rep1_Aligned.sortedByCoord.out.bam | more
----

or at the mapping statistics that come with `STAR`:

[source,cmd,subs="{markup-in-source}"]
----
cat alignments/mouse_cns_E18_rep1_Log.final.out
----
----
                                 Started job on |	Oct 22 09:47:38
                             Started mapping on |	Oct 22 09:47:38
                                    Finished on |	Oct 22 09:48:19
       Mapping speed, Million of reads per hour |	8.78

                          Number of input reads |	100000
                      Average input read length |	202
                                    UNIQUE READS:
                   Uniquely mapped reads number |	90153
                        Uniquely mapped reads % |	90.15%
                          Average mapped length |	200.44
                       Number of splices: Total |	36918
            Number of splices: Annotated (sjdb) |	121
                       Number of splices: GT/AG |	36559
                       Number of splices: GC/AG |	190
                       Number of splices: AT/AC |	2
               Number of splices: Non-canonical |	167
                      Mismatch rate per base, % |	0.23%
                         Deletion rate per base |	0.01%
                        Deletion average length |	1.59
                        Insertion rate per base |	0.00%
                       Insertion average length |	1.39
                             MULTI-MAPPING READS:
        Number of reads mapped to multiple loci |	1720
             % of reads mapped to multiple loci |	1.72%
        Number of reads mapped to too many loci |	100
             % of reads mapped to too many loci |	0.10%
                                  UNMAPPED READS:
       % of reads unmapped: too many mismatches |	0.00%
                 % of reads unmapped: too short |	7.96%
                     % of reads unmapped: other |	0.06%
                                  CHIMERIC READS:
                       Number of chimeric reads |	0
                            % of chimeric reads |	0.00%
----
