= Transcript and gene expression quantification

Create a folder for the quantifications:

[source,bash]
----
mkdir quantifications
----

Create a bash script called `run_rsem.sh` with the following:

[source,bash]
----
#!/bin/bash -e

# load env
. ~/rnaseq/.rnaseqenv

# load module
module load RSEM/1.2.21-goolf-1.4.10-no-OFED

# get quantifications with RSEM
rsem-calculate-expression --bam --estimate-rspd --no-bam-output --seed 12345 -p 2 --paired-end --forward-prob 0 alignments/mouse_cns_E18_rep1_Aligned.toTranscriptome.out.bam ~/rnaseq/refs/mouse_genome_mm9_RSEM_index/RSEMref quantifications/mouse_cns_E18_rep1
----

Submit the job to the cluster:

[source,cmd]
----
qsub -cwd -q RNAseq -l virtual_free=16G -pe smp 2 -N isoforms_rnaseq_course -e logs -o logs ./run_rsem.sh
----

To obtain a matrix of gene FPKM values:

[source,cmd]
----
cat ~/rnaseq/data/quantifications.index.txt | retrieve_element_rpkms.py -o encode -O mouse -e gene -v FPKM -d quantifications
----

To obtain a matrix of transcript FPKM values:

[source,cmd]
----
cat ~/rnaseq/data/quantifications.index.txt | sed 's/gene/isoform/' | retrieve_element_rpkms.py -o encode -O mouse -e transcript -v FPKM -d quantifications
----

To obtain a matrix of gene read counts:

[source,cmd]
----
cat ~/rnaseq/data/quantifications.index.txt | retrieve_element_rpkms.py -o encode -O mouse -e gene -v expected_count -d quantifications
----

To obtain a matrix of transcript read counts:

[source,cmd]
----
cat ~/rnaseq/data/quantifications.index.txt | sed 's/gene/isoform/' | retrieve_element_rpkms.py -o encode -O mouse -e transcript -v expected_count -d quantifications
----
