= Transcript and gene expression quantification

Create a folder for the quantifications:

[source,cmd]
----
mkdir quantifications
----

Create the transcriptome index for RSEM:

[source,cmd,subs="{markup-in-source}"]
----
mkdir txIndex
----

[source,cmd,subs="{markup-in-source}"]
----
rsem-prepare-reference --gtf {rnaseq_dir}refs/anno.gtf \
                      {rnaseq_dir}refs/genome.fa \
                      txIndex/RSEMref
----

Run the following command to estimete the expression:

[source,cmd,subs="{markup-in-source}"]
----
rsem-calculate-expression -p 2 \
                          --bam \
                          --paired-end \
                          --estimate-rspd \
                          --forward-prob 0 \
                          --no-bam-output \
                          --seed 12345 \
                          alignments/mouse_cns_E18_rep1_Aligned.toTranscriptome.out.bam \
                          txIndex/RSEMref \
                          quantifications/mouse_cns_E18_rep1
----

NOTE: Here we are correctly treating the data as [crg]#**stranded**#.

RSEM generates expression quantification at transcript and gene level. For the data analysis section we are going to use only [crg]#gene expression data#.

To obtain a matrix of gene FPKM values:

[source,cmd,subs="{markup-in-source}"]
----
cat {rnaseq_folder}data/quantifications.index.txt | retrieve_element_rpkms.py -o encode -O mouse -e gene -v FPKM -d quantifications
----

To obtain a matrix of gene read counts:

[source,cmd,subs="{markup-in-source}"]
----
cat {rnaseq_folder}data/quantifications.index.txt | retrieve_element_rpkms.py -o encode -O mouse -e gene -v expected_count -d quantifications
----
