= Transcript and gene expression quantification

Create a folder for the quantifications:

[source,cmd]
----
mkdir quantifications
----

Create the transcriptome index for RSEM:

[source,cmd,subs="{markup-in-source}"]
----
mkdir txIndex
----

[source,cmd,subs="{markup-in-source}"]
----
rsem-prepare-reference --gtf {rnaseq_folder}refs/anno.gtf \
                      {rnaseq_folder}refs/genome.fa \
                      txIndex/RSEMref
----
// real	0m0.420s
// user	0m0.371s
// sys	0m0.049s

Run the following command to estimete the expression:

[source,cmd,subs="{markup-in-source}"]
----
rsem-calculate-expression --num-threads 2 \
                          --bam \
                          --paired-end \
                          --estimate-rspd \
                          --forward-prob 0 \
                          --no-bam-output \
                          --seed 12345 \
                          alignments/mouse_cns_E18_rep1_Aligned.toTranscriptome.out.bam \
                          txIndex/RSEMref \
                          quantifications/mouse_cns_E18_rep1
----
// real	0m5.070s
// user	0m7.821s
// sys	0m0.597s

NOTE: Here we are correctly treating the data as [crg]#**stranded**#.

NOTE: Remember that, in general, you can always have a look at programs options by typing `--help` after the command (e.g. `rsem-calculate-expression --help`).

RSEM generates expression quantification at transcript and gene level. For the data analysis section we are going to use only [crg]#gene expression data#.

To obtain a matrix of gene FPKM values:

[source,cmd,subs="{markup-in-source}"]
----
cat {rnaseq_folder}data/quantifications.index.txt \
| retrieve_element_rpkms.py --output encode \
                            --organism mouse \
                            --element gene \
                            --value FPKM \
                            --output_dir quantifications
----

To obtain a matrix of gene read counts:

[source,cmd,subs="{markup-in-source}"]
----
cat {rnaseq_folder}data/quantifications.index.txt \
| retrieve_element_rpkms.py --output encode \
                            --organism mouse \
                            --element gene \
                            --value expected_count \
                            --output_dir quantifications
----
