= GO enrichment

Prepare a file with the list of all the genes in the annotation:

[source,cmd,subs="{markup-in-source}"]
----
awk '{split($10,a,/\"|\./); print a[2]}' {rnaseq_folder}refs/gencode.vM4.gtf | sort -u > universe.txt
----

Launch the GO enrichment script for the Biological Processes, Molecular Function and Cellular Components in the set of genes overexpressed in brain:

[source,cmd]
----
# BP
awk '{split($1,a,"."); print a[1]}' edgeR.0.01.over_brain_X_heart.txt \
| GO_enrichment.R --universe universe.txt \
                  --genes stdin \
                  --categ BP \
                  --output edgeR.over_brain_X_heart \
                  --species mouse

# MF
awk '{split($1,a,"."); print a[1]}' edgeR.0.01.over_brain_X_heart.txt \
| GO_enrichment.R --universe universe.txt \
                  --genes stdin \
                  --categ MF \
                  --output edgeR.over_brain_X_heart \
                  --species mouse

# CC
awk '{split($1,a,"."); print a[1]}' edgeR.0.01.over_brain_X_heart.txt \
| GO_enrichment.R --universe universe.txt \
                  --genes stdin \
                  --categ CC \
                  --output edgeR.over_brain_X_heart \
                  --species mouse
----

The results can be visualized in a web browser, using the online Gene Ontology visualizer link:http://revigo.irb.hr[REVIGO^]. As an example, prepare REVIGO input using Biological Process (BP) results:

[source,cmd]
----
awk 'NR==1{$1="% "$1}{print $1,$2}' edgeR.over_brain_X_heart.BP.tsv
----

Copy the above command output and paste it in the text box on the main page of the REVIGO web site. Then click the button btn:[Start Revigo] and have a look at the results.

You can repeat the same for the genes overexpressed in heart.
