= RNA-seq data analysis
:experimental:
:UCSC_genome_browser: http://genome.ucsc.edu

Create a directory dedicated to the analyses:

[source,bash]
----
mkdir analysis
----

And move into it:

[source,bash]
----
cd analysis
----

Load the environment:

[source,bash]
----
. ~/rnaseq/.rnaseqenv
----

== RPKM distribution

Have a look at the distribution of RPKM values:

[source,bash]
----
rpkm_distribution.R -i ../quantifications/encode.mouse.gene.FPKM.idr_NA.tsv -l -p 0 -m ../data/quantifications.index.tsv -f age
----

To look at the plot:

[source,bash]
----
evince boxplot.log_T.psd_0.out.pdf
----

== Clustering analysis

Perform hierarchical clustering to check replicability:

[source,bash]
----
matrix_to_dist.R -i ../quantifications/encode.mouse.gene.FPKM.idr_NA.tsv --log10 -c pearson -o stdout | ggheatmap.R -i stdin --row_metadata ../data/quantifications.index.tsv --col_dendro --row_dendro -B 10 --matrix_palette=~/rnaseq/bin/terrain.colors.3.txt --rowSide_by age --matrix_fill_limits 0.85,1 -o cns.heatmap.pdf
----

Look at the clustering.

== Differential gene expression

Run the DE with the edgeR package (be careful takes read counts and not rpkm values as input):

[source,bash]
----
edgeR.analysis.R -i ../quantifications/encode.mouse.gene.expected_count.idr_NA.tsv -m ../data/quantifications.index.tsv -f age
----

Write a list of the genes overexpressed after 18 days, according to edgeR analysis:

[source,bash]
----
awk '$NF<0.01 && $4>2{print $1"\tover18"}' edgeR.cpm1.n2.out.tsv > edgeR.0.01.overE18.txt
----

Write a list of the genes overexpressed after 14 days, according to edgeR analysis:

[source,bash]
----
awk '$NF<0.01 && $4<-2 {print $1"\tover14"}' edgeR.cpm1.n2.out.tsv > edgeR.0.01.overE14.txt
----

Count how many overexpressed genes there are in each stage:

[source,bash]
----
wc -l edgeR.0.01.over*.txt
----

Show the results in a heatmap:

[source,bash]
----
(echo -e "gene\tedgeR"; cat edgeR.0.01.over*.txt) > gene.edgeR.tsv
cut -f1 gene.edgeR.tsv | tail -n+2 | selectMatrixRows.sh - ../quantifications/encode.mouse.gene.FPKM.idr_NA.tsv | ggheatmap.R -W 5 -H 9 --col_metadata ../data/quantifications.index.tsv --colSide_by age --col_labels labExpId --row_metadata gene.edgeR.tsv --merge_row_mdata_on gene --rowSide_by edgeR --row_labels none -l -p 0.1 --col_dendro --row_dendro -o heatmap.edgeR.pdf
----

== GO enrichment

Prepare a file with the list of all the genes in the annotation:

[source,bash]
----
awk '{split($10,a,"\""); print a[2]}' ~/rnaseq/refs/mm65.long.ok.gtf | sort -u > universe.txt
----

Launch the GO enrichment script for the Biological Processes, Molecular Function and Cellular Components in the set of genes overexpressed in E14:

[source,bash]
----
cut -f1 edgeR.0.01.overE14.txt | GO_enrichment.R -u universe.txt -G stdin -c BP -o edgeR.overE14 -s mouse
cut -f1 edgeR.0.01.overE14.txt | GO_enrichment.R -u universe.txt -G stdin -c MF -o edgeR.overE14 -s mouse
cut -f1 edgeR.0.01.overE14.txt | GO_enrichment.R -u universe.txt -G stdin -c CC -o edgeR.overE14 -s mouse
----

The results can be visualized in the browser, pasting the following paths in the search line:

[source,bash]
----
firefox ~/rnaseq/analysis/edgeR.overE14.BP.html
firefox ~/rnaseq/analysis/edgeR.overE14.MF.html
firefox ~/rnaseq/analysis/edgeR.overE14.CC.html
----

You can repeat the same for the genes overexpressed in E18:

[source,bash]
----
cut -f1 edgeR.0.01.overE18.txt | GO_enrichment.R -u universe.txt -G stdin -c BP -o edgeR.overE18 -s mouse
cut -f1 edgeR.0.01.overE18.txt | GO_enrichment.R -u universe.txt -G stdin -c MF -o edgeR.overE18 -s mouse
cut -f1 edgeR.0.01.overE18.txt | GO_enrichment.R -u universe.txt -G stdin -c CC -o edgeR.overE18 -s mouse
----
